### ISSUE TITLE
SET commands in headless and interactive differ in behavior

### ISSUE THREAD

**Issue by agavra** [confluentinc/ksql#4930]
**Describe the bug**

It seems that since 5.4 some `SET` commands cause headless mode to fail, but work in interactive mode.

**To Reproduce**

Run a query file with `SET 'max.task.idle.ms'='100000'` and some queries

**Expected behavior**

The above should succeed and set the property.

**Actual behaviour**

```
[2020-03-18 01:52:15,421] ERROR Failed to start KSQL (io.confluent.ksql.rest.server.KsqlServerMain:61)
io.confluent.ksql.util.KsqlStatementException: Unknown property: max.task.idle.ms
Statement: SET 'max.task.idle.ms' = '100000';
```

**Comment by agavra** [comment]
This is caused because `SET` commands in interactive mode are not distributed, instead they just set your local CLI properties (see: [SetCliProperty.java](https://github.com/confluentinc/ksql/blob/master/ksqldb-cli/src/main/java/io/confluent/ksql/cli/console/cmd/SetCliProperty.java)) and then are distributed using the property overrides. 

On the other hand, the headless mode directly goes through `PropertyOverrider#set` which validates that the properties are within a set of specific whitelisted properties.

My recommendation would be to unify the path - i.e. to remove the set of validated configs introduced by https://github.com/confluentinc/ksql/pull/2618 (see discussion on the PR) and to go through the same validator that the CLI uses.

**Comment by big-andy-coates** [comment]
So.... is it the case that while the `SET 'max.task.idle.ms'='100000'` doesn't fail in headless, it's not actually doing anything, as there is something wrong with the name or value?

**Comment by agavra** [comment]
> So.... is it the case that while the `SET 'max.task.idle.ms'='100000'` doesn't fail in headless, it's not actually doing anything, as there is something wrong with the name or value?

did you mean doesn't fail in "interactive"? it doesn't fail because it can circumvent the `PropertyExecutor`. Instead, it just goes through the overrides that are sent alongside each statement (`propertyOverrides`) so I think it _does_ actually do something.

**Comment by vcrfxia** [comment]
Closing since https://github.com/confluentinc/ksql/pull/4936 has been merged. The fix will be included in 5.4.2, 5.5.0+, and 0.9.0+
